module/migration.mjs

```diff
@@ -13,11 +13,13 @@ export const migrateWorld = async function() {
     .concat(Array.from(game.actors.invalidDocumentIds).map(id => [game.actors.getInvalid(id), false]));
   for ( const [actor, valid] of actors ) {
     try {
+      const flags = { needsMigration: false };
       const source = valid ? actor.toObject() : game.data.actors.find(a => a._id === actor.id);
-      const updateData = migrateActorData(source, migrationData);
-      if ( !foundry.utils.isEmpty(updateData) ) {
+      const updateData = migrateActorData(source, migrationData, flags);
+      if ( !foundry.utils.isEmpty(updateData) || flags.needsMigration ) {
         console.log(`Migrating Actor document ${actor.name}`);
-        await actor.update(updateData, {enforceTypes: false, diff: valid});
+        const update = foundry.utils.isEmpty(updateData) ? source : updateData;
```

Comment:
The full source data needs to take priority in the 'needsMigration' case, otherwise we might not persist the source migration in cases where there's both a source migration and a non-source migration.

```suggestion
        if ( flags.needsMigration ) foundry.utils.mergeObject(source, updateData);
        const update = flags.needsMigration ? source : updateData;
```